lib/owners/lexer.l

```diff
@@ -5,7 +5,7 @@ macro
   TEAMNAME          @[a-zA-Z0-9\-]+\/[a-zA-Z0-9\-]+
   USERNAME          @[a-zA-Z0-9\-]+
   GLOB              [a-zA-Z0-9_\-\*\?\.]+
-  COMMENT           \# [^\"\n\"]*
+  COMMENT           \#\s[^\"\n\"]*
```

Comment:
I thought about this a bit more, and I think there is another edge case to consider. As it stands `\s` matches exactly one piece of whitespace, including the newline character. I think it should probably be `[\t ]` instead.

Example:

```
irb(main):003:0> Owners::Parser.new.parse("#\n@team")
=> [[:COMMENT, "#\n@team"], [:END, nil]]
```

The `\n` is consumed by `\s`, meaning the `@team` on the next line is considered part of the comment.

A few other things I thought about, but don't know the answer to:
1. Why are double quotes not allowed, and why is double quote character listed twice in the character class? Having a character listed multiple times within `[ ]` has no effect, as far as I know, so I'm wondering if it's possible the original author had a different intention than what this code actually does, or if it's just a typo/mistake.
2. Should the whitespace between `#` and the start of the comment be optional? That is, should we support `#comment` or would it have a negative impact on other parts of the parser that I am less familiar with?